src/index.js

```diff
@@ -207,28 +207,23 @@ export default function undoable (reducer, rawConfig = {}) {
     clearHistoryType: rawConfig.clearHistoryType || ActionTypes.CLEAR_HISTORY
   }
 
-  return (state, action = {}) => {
+  return (state = config.history, action = {}) => {
     debugStart(action, state)
 
-    let history
+    let history = state
     if (!config.history) {
-      debug('create history on init')
+      debug('history is uninitialized')
 
       if (state === undefined) {
-        config.history = createHistory(reducer(state, {}))
+        history = createHistory(reducer(state, {}))
+        debug('do not initialize on probe actions')
       } else if (isHistory(state)) {
-        config.history = state
+        history = config.history = state
+        debug('initialHistory properly shaped and initialized', config.history)
       } else {
-        config.history = createHistory(state)
+        history = config.history = createHistory(state)
+        debug('initialHistory reshaped and initialized', config.history)
```

Comment:
I'm fine with rephrasing this to something else. If this statement executes then `createStore` was passed an `initialState` which was not shaped like a `history` object. So when I say "reshaped", I meant to say that a normal object was used to create a history with `createHistory`. Perhaps these two messages are more clear:
```js
-- snip --
  } else if (isHistory(state)) {
    history = config.history = state
    debug('initialHistory initialized: initialState is a history', config.history)
  } else {
    history = config.history = createHistory(state)
    debug('initialHistory initialized: initialState is not a history', config.history)
  }
```